src/tyr/route_summary_cache.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
```

Comment:
i wasnt suggesting adding it as a public header i was suggesting taking this and the implementation and putting it in a header inside of the source directory and including it only from the serializer (private header). i think its good to keep implementation details like this out of the top level apis (since we already have a ton of stuff there)

Reply:
Hmm.  I saw the route_summary_cache as only ever meant to be utilized by one code consumer - hence why it was implemented in the source file.

Moving it into a header implies the possibility that another code consumer could use it.  I didn't really see it being the case, but that's how I read your suggestion.  In which case, the methods were too big to justify being inlined, hence why I out-of-lined them.

But this comes down to a design preference and I'm starting to read between the lines.  I think you're suggesting that the the route_summary_cache 1) still just have one code customer, 2) be fully inlined which is in effect the same as having the route_summary_cache implemented in the source, but 3) allows the code to be modularized into its own home.

All that being said, I'm glad to take your suggestion.  Following existing design precedents is important to me.

Reply:
yeah sorry i didnt spell out why i was asking that very well, apologies for that. number 3 is probably the main reason why i suggested it. giving it a small footprint but letting it be sequestered into its own area seems important to me. it also makes it really straight forward to unit test directly.

Reply:
thanks, suggestion taken.